src/utils/resources/vmi/utils/cloud-init.ts

```diff
@@ -1,23 +1,38 @@
+import { load } from 'js-yaml';
+
 import { V1VirtualMachineInstance } from '@kubevirt-ui/kubevirt-api/kubevirt';
 
+type CloudinitUserDataObject = {
+  user?: string;
+  password?: string;
+  passwd?: { users: { name: string }[] };
+};
+
 export const getCloudInitCredentials = (
   vmi: V1VirtualMachineInstance,
-): { user: string; password: string } => {
+): { user?: string; password?: string; users?: string[] } => {
   const cloudInitVolume = vmi?.spec?.volumes?.find(
     (volume) => volume?.cloudInitNoCloud || volume?.cloudInitConfigDrive,
   );
 
-  const cloudInit = cloudInitVolume?.cloudInitNoCloud || cloudInitVolume?.cloudInitConfigDrive;
+  // KubeVirt supports the cloud-init NoCloud and ConfigDrive data sources
+  const cloudInitDataSource =
+    cloudInitVolume?.cloudInitNoCloud || cloudInitVolume?.cloudInitConfigDrive;
 
-  const cloudInitValuesStrings = cloudInit?.userData
-    ?.split('\n')
-    ?.filter((row) => !row?.includes('#cloud-config'));
+  const userDataObject: CloudinitUserDataObject = load(cloudInitDataSource?.userData);
```

Comment:
I think its missing the try-catch block as the load can throw an exception when failed to parse